src/adapters/store.js

```diff
@@ -18,31 +18,29 @@ function Store() {
   })
 }
 
-Store.prototype.getAll = async function() {
-  return new Promise((resolve, reject) => {
-    abstractStore.getAll().then((masqData) => {
-      resolve(masqData)
-    }).catch(function (error) {
-      fire('error_h' , 'store ' + error)
-      reject(error)
-    })
-  })
+Store.prototype.getAllPois = async function() {
+  try {
+    let masqData = await abstractStore.getAllPois()
+    return masqData
+  } catch (error) {
+    fire('error_h' , 'store ' + error)
+    throw error
+  }
 }
 
 Store.prototype.isRegistered = async function () {
-  return new Promise((resolve) => {
-    abstractStore.getAll()
-      .then(() => resolve(true))
-      .catch((e) => {
-      if(e.message === 'UNREGISTERED') {
-        resolve(false)
-      }
-    })
-  })
+  try {
+    let data = await abstractStore.getAllPois()
+    return data != null
+  } catch (error) {
+    if(error.message === 'UNREGISTERED') {
+      return false
+    }
```

Comment:
this method let an possible unsolvable promise (in the case of error.message != 'unregistred') maybe we can delete the message test and return false every time we enter in the catch.